modules/ServerAPI/src/client/Protocol.coffee

```diff
@@ -499,3 +530,560 @@ class ProtocolBaseController extends BaseEntityController
+			rnB = @getRowNumberForState(stateB)
+			return rnA - rnB
+		
+	render: => 
+		$(@el).empty()
+		$(@el).html @template()
+
+		#If the table is read-only, remove the remove and add buttons
+		if @options.readOnly == true
+			#remove column
+			@$(".bv_endpointColumnRemove").remove()
+			#remove add endpoint button
+			@$(".bv_addEndpoint").remove()
+			$(".bv_endpointManagerInstructions").hide()
+		
+		if @options.newProtocol == true
+			@$(".bv_downloadFiles").hide()
+			$(".bv_endpointManagerInstructions").hide()
+
+		#if time/units or concentration/units are disabled, remove the columns from the endpoint coontroller
+		if window.conf.protocol.endpointManager.showTime == false
+			@$(".bv_endpointColumnTime").remove()
+			@$(".bv_endpointColumnTimeUnits").remove()
+
+		if window.conf.protocol.endpointManager.showConcentration == false
+			@$(".bv_endpointColumnConcentration").remove()
+			@$(".bv_endpointColumnConcentrationUnits").remove()
+		
+		#Placeholder until we update the protocol data structure
+		# Create a list to hold the endpoint controllers in, so we can iterate through them later
+		@endpointControllers = []
+		for lsState in @collection
+			@.addOne(lsState)
+			if @options.readOnly == true
+				#hide remove buttons
+				@$(".bv_remove_row").hide()
+
+
+		if @options.readOnly == false || @options.newProtocol == true 	#Don't render associated experiments if protocol is new or readOnly
+			@getExperimentSummaryTable()
+		else #if the table isn't rendered, don't render the download files button either
+			$(".bv_downloadFiles").hide() 
+
+		@
+	
+	getExperimentSummaryTable: =>
+		#hide previously shown warnings/success text
+		$(".bv_downloadSuccess").hide()
+		$(".bv_downloadWarning").hide()
+
+		protocolCode = @model.escape('codeName')		
+		$.ajax
+			type: 'GET'
+			#there are two similar routes in ExperimentBrowserRoutes.coffee and ExperimentServiceRoutes.coffee
+			#url: "/api/experimentsForProtocol/#{protocolCode}" #ExperimentBrowserRoutes.coffee route
+			url: "/api/experiments/protocolCodename/#{protocolCode}" #ExperimentServiceRoutes.coffee route
+			success: (experiments) =>
+				@setupExperimentSummaryTable experiments
+				$(".bv_experimentTableControllerTitle").html "Experiments using " + protocolCode + ":"
+
+
+	handleEndpointRowPressed: =>
+		if @options.readOnly == false #disable logic if endpoint list controller is read only (since there won't be an experiment controller)
+			#When an endpoint row is pressed, we want to filter the experiment summary table so it only displays the experiments...
+			#...with protocols that contain the endpoint, so we'll need to find the row's endpoint name and regenerate the table
+
+			#since any element within the row could be clicked on, we need to first find the parent row div and move down from there
+			element = $(event.target)
+			for x in [0..20] #allow it to search up to 20 parent elements up for the table row (usually takes 10 or less)
```

Comment:
This is clever, but I think you can use a library to do it more succinctly. Could you use `.closest('tr');` to do this?
https://stackoverflow.com/questions/9314902/jquery-get-parent-tr-for-selected-radio-button

Reply:
Thats super helpful, thanks!